### ISSUE TITLE
Infinite loop selecting reachability of an edge after starting simulation

### ISSUE THREAD

**Issue by palvarezlopez** [eclipse/sumo#12400]
Found in SUMO-GUI 1.15
[SUMOConfigs.zip](https://github.com/eclipse/sumo/files/10361033/SUMOConfigs.zip)

- Open sumo config
- Start and stop simulation
- Select edge reachability (see image)
- Infinite loop and crash

![Sin título](https://user-images.githubusercontent.com/8238561/211039873-56c75ce3-ecb4-4ebc-941c-4c00e9aaa100.png)

The problem is probably caused by FXMenuCommand* mc = dynamic_cast<FXMenuCommand*>(menu). This dynamic_cast<FXMenuCommand*> can be null, and could provoke an error.

Solution: Avoid to use the dynamic cast.
```
long
GUIApplicationWindow::onCmdEditChosen(FXObject* menu, FXSelector, void*) {
    FXMenuCommand* mc = dynamic_cast<FXMenuCommand*>(menu);
    if (mc->getText() == "Edit Selected...") {
        GUIDialog_GLChosenEditor* chooser =
            new GUIDialog_GLChosenEditor(this, &gSelected);
        chooser->create();
        chooser->show();
    } else {
        if (!myAmLoading && myRunThread->simulationAvailable()) {
            const SUMOVehicleClass svc = SumoVehicleClassStrings.get(mc->getText().text());
            for (MSEdgeVector::const_iterator i = MSEdge::getAllEdges().begin(); i != MSEdge::getAllEdges().end(); ++i) {
                const std::vector<MSLane*>& lanes = (*i)->getLanes();
```

**Comment by namdre** [comment]
was last working in 1.13.0. 
In 1.14.0 and 1.14.1 it did nothing (but at least it didn't crash)

**Comment by namdre** [comment]
- The problem from 1.14.0 was already fixed in 1.15.0 (#11448)
- The infinite loop is a network specific problem that has been present since the first inception of the funcionality

**Comment by namdre** [comment]
remaining work continues in #12403
